ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -187,28 +190,79 @@ internal constructor(
     externalScope.launch(ioDispatcher) { submitDataUseCase.invoke(loiId, job, surveyId, deltas) }
   }
 
-  /** Returns the position of the task fragment visible to the user. */
-  fun getVisibleTaskPosition() = currentPosition.value
-
-  /** Displays the task at the given position to the user. */
-  fun updateCurrentPosition(position: Int) {
-    savedStateHandle[TASK_POSITION_KEY] = position
+  fun getAbsolutePosition(): Int {
+    if (currentTaskId.value == "") {
+      return 0
+    }
+    return tasks.indexOf(tasks.first { it.id == currentTaskId.value })
   }
 
-  /** Returns true if the given task position is last. */
-  fun isLastPosition(taskPosition: Int): Boolean {
-    val finalTaskPosition = tasks.size - 1
+  /** Get the index of the current task within the (possibly conditional) task sequence. */
+  fun getRelativePosition(): Pair<Int, Int> {
```

Comment:
What's do the values in the return Pair correspond to?

Also, what is RelativePosition relative to, any why do we need this? Is there a way to make that clear from code?